### PR TITLE
Mzieg multichannel 2

### PR DESCRIPTION
**Pull request by mzieg**

### CODE REVIEW

File: MultiChannelDemo/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -32,5 +32,5 @@
 // You can specify all the values or you can default the Build and Revision Numbers
 // by using the '*' as shown below:
 // [assembly: AssemblyVersion("1.0.*")]
-[assembly: AssemblyVersion("2.2.1.0")]
-[assembly: AssemblyFileVersion("2.2.1.0")]
+[assembly: AssemblyVersion("2.2.5.0")]
```
Comment (mzieg) [inline MultiChannelDemo/Properties/AssemblyInfo.cs:35]: will adjust at merge

File: README_CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,11 @@
 # Wasatch.NET Changelog
 
+- 2020-05-?? 2.2.?
```
Comment (mzieg) [inline README_CHANGELOG.md:3]: will fix version numbers at merge

File: WasatchNET/EEPROM.cs [hunk 3]
```diff
@@ -30,7 +30,7 @@ public class EEPROM : IEEPROM
 
         //internal const int MAX_PAGES = 256; // really 8, but last 2 are unallocated
         internal const int MAX_PAGES = 8;
-        const byte FORMAT = 8;
+        const byte FORMAT = 9;
```
Comment (mzieg) [inline WasatchNET/EEPROM.cs:33]: This should get your attention.

File: MultiChannelDemo/Form1.cs [hunk 4]
```diff
@@ -1163,5 +863,25 @@ private void backgroundWorkerBatch_RunWorkerCompleted(object sender, RunWorkerCo
...
+
+        void updateScanAveragingAvailability()
+        {
+            if (wrapper.hardwareTriggeringEnabled)
+                numericUpDownScansToAverage.Enabled = wrapper.useContinuousAcquisition;
+            else
+                numericUpDownScansToAverage.Enabled = true;
+        }
     }
 }
```
Comment (trestoh) [inline MultiChannelDemo/Form1.cs:887]: Now I know how you must've felt looking at wpsc 2.0....

File: WasatchNET/Spectrometer.cs [hunk 5]
```diff
@@ -2209,7 +2327,7 @@ protected virtual double[] getSpectrumRaw()
                 {
                     if (!currentAcquisitionCancelled && errorOnTimeout)
                         logger.error($"failed when reading subspectrum from 0x{spectralReader.EpNum:x2} ({id})");
-                    Thread.Sleep(10);
+                    Thread.Sleep(100);
```
Comment (trestoh) [inline WasatchNET/Spectrometer.cs:2330]: My inclination is this should be configurable

File: WasatchNET/Spectrometer.cs [hunk 6]
```diff
@@ -2083,6 +2191,10 @@ public virtual double[] getSpectrum(bool forceNew=false)
                     for (int px = 0; px < pixels; px++)
                         sum[px] -= dark_[px];
 
+                // this should be enough to update the cached value
+                if (readTemperatureAfterSpectrum)
+                    _ = detectorTemperatureDegC;
+
```
Comment (trestoh) [inline WasatchNET/Spectrometer.cs:2197]: where is the cache being set? Shouldn't it be here? Might be missing a cache in detectorTemperatureDegC's setter

Comment (trestoh) [reply WasatchNET/Spectrometer.cs:2197]: if so disregard

Comment (mzieg) [reply WasatchNET/Spectrometer.cs:2197]: It's set in the last line of the Spectrometer.detectorTemperatureDegC getter.

File: WasatchNET/Spectrometer.cs (hunk 5 above)
Comment (mzieg) [reply WasatchNET/Spectrometer.cs:2330]: Done.

**Pull request merged**